pkg/cli/admin/release/info.go

```diff
@@ -328,19 +335,22 @@ func replaceClusterSemanticArgs(f kcmdutil.Factory, args []string, semanticArgs
 	return args, nil
 }
 
-func findArgumentsFromCluster(f kcmdutil.Factory, args []string) ([]string, error) {
+func findArgumentsFromCluster(f kcmdutil.Factory, args []string, arch string) ([]string, error) {
 	semanticArgs := findSemanticVersionArgs(args)
 	if len(semanticArgs) == 0 && len(args) > 0 {
 		return args, nil
 	}
 	klog.V(4).Infof("Found semantic versions: %v", semanticArgs)
 	// attempt to find semantic args from the cluster
-	args, clusterErr := replaceClusterSemanticArgs(f, args, semanticArgs)
+	args, clusterErr := replaceClusterSemanticArgs(f, args, semanticArgs, arch)
+
+	// TODO compare the arch from the returned release image from cluster to the value passed from --release-arch.. if different get release image from api?
+
```

Comment:
@Prashanth684 I do need some help here.. The way the current look up sequence is:

When connected to a cluster:

oc adm release info will return that clusters release container and its contents
oc adm release info 4.x.x will return that clusters release container ONLY if the release version matches the value passed. If the arch value isn't passed it will return x86
So my question is do we want oc adm relase info --graph-arch=<arch different than the cluster> to reach to the graph api endpoint and pull the right payload? I think this is doable, but not seeing an easy/obvious why to get a clusters architecture w/o cracking open the release container?

Reply:
so this is my thinking:
- if `--graph-arch` or `--release-arch` whatever we call it is specified, there has to be a release version specified, if not we error out
- if the option is specified along with a release version, we always fetch from the graph even if the cluster has the same version.
- if a release version is specified without the arch option, then it can get it from the release container if the release version matches the cluster version

imo it doesn't make sense to inspect the cluster's arch (which can be done by looking at a node's kubernetes.io/arch label at least for homogeneous clusters). i would assume that if a user specifies the arch option, they want to know the release for a particular arch and not the clusters'.